### PR TITLE
upgrading nixpkgs

### PR DESCRIPTION
**Pull request by CohenCyril**

### CODE REVIEW

**Comment by amahboubi** [comment]
What is the upgrade about?

**Comment by CohenCyril** [comment]
This is the equivalent of `opam upgrade` except it does so for all users. Be prepared to recompile mathcomp if we merge this.

**Comment by amahboubi** [comment]
How do we know the nature of the upgrade (from which version to which other)?

**Comment by CohenCyril** [comment]
> How do we know the nature of the upgrade (from which version to which other)?

I know no other way than running nix-shell and observing the computed set of packages to install. (Note that this does not uninstall the previous versions, which remain available if you change the commit back to the previous one)

**Comment by amahboubi** [comment]
I meant, for the present commit log, and for analysis, which version of the dependencies are supported by this new config. E.g. real-closed is now available, etc.

**Comment by CohenCyril** [comment]
> I meant, for the present commit log, and for analysis, which version of the dependencies are supported by this new config. E.g. real-closed is now available, etc.

cf https://github.com/NixOS/nixpkgs/blob/73392e79aa62e406683d6a732eb4f4101f4732be/pkgs/development/coq-modules/mathcomp/extra.nix#L46-L61

**Comment by amahboubi** [comment]
Could you rephrase your commit log to reflect the upgrades it allows?

**Comment by amahboubi** [comment]
> cf https://github.com/NixOS/nixpkgs/blob/73392e79aa62e406683d6a732eb4f4101f4732be/pkgs/development/coq-modules/mathcomp/extra.nix#L46-L61

Could we have a nix upgrade documentation somewhere? E..g. what is the role of this file? Who should edit it and when?

**Comment by CohenCyril** [comment]
> Could we have a nix upgrade documentation somewhere? E..g. what is the role of this file? Who should edit it and when?

a doc on
- upgrading the local nix shell (shell.nix) on analysis?
- or upgrading the package definitions (mathcomp/extra.nix) on nixpkgs?

**Comment by amahboubi** [comment]
Both.

**Comment by amahboubi** [comment]
@CohenCyril , again a description would have been really helpful here, listing the packages and/or versions made available via this update.
Also, such a PR should not say un-merged that long (without a reason).
